src/structures/GuildPreview.js

```diff
@@ -0,0 +1,171 @@
+'use strict';
+
+const Base = require('./Base');
+const GuildEmojiManager = require('../managers/GuildEmojiManager');
+
+/**
+ * Represents the data about the guild any bot can preivew, connected to the specified public guild
+ */
+class GuildPreview extends Base {
+  constructor(client, data) {
+    super(client);
+
+    if (!data) return;
+    this._patch(data);
+  }
+
+  /**
+   * Builds the public guild with the provided data
+   * @param {*} data The raw data of the public guild
+   * @private
+   */
+  _patch(data) {
+    /**
+     * The id of this public guild
+     * @type {string}
+     */
+    this.id = data.id;
+
+    /**
+     * The name of this public guild
+     * @type {string}
+     */
+    this.name = data.name;
+
+    /**
+     * The icon of this public guild
+     * @type {?string}
+     */
+    this.icon = data.icon;
+
+    /**
+     * The splash icon of this public guild
+     * @type {?string}
+     */
+    this.splash = data.splash;
+
+    /**
+     * The discovery splash icon of this public guild
+     * @type {?string}
+     */
+    this.discoverySplash = data.discovery_splash;
+
+    /**
+     * An array of enabled guild features, here are the possible values:
+     * * ANIMATED_ICON
+     * * BANNER
+     * * COMMERCE
+     * * DISCOVERABLE
+     * * FEATURABLE
+     * * INVITE_SPLASH
+     * * PUBLIC
+     * * NEWS
+     * * PARTNERED
+     * * VANITY_URL
+     * * VERIFIED
+     * * VIP_REGIONS
+     * @typedef {string} Features
```

Comment:
i believe a typedef already exists for this

Reply:
Yes, it does. I apologize, I forgot to remove that when comitting. Thank you for pointing it out.